src/lib.rs

```diff
@@ -150,19 +251,6 @@ impl<'s, T: ToString, Z: ToString> EnvironmentItem for &'s (T, Z) {
     }
 }
 
-impl<'s, T> From<T> for Environment
```

Comment:
Why did you get rid of this?  Its a lot more generalized way of handling converting to an Environment

Reply:
Because I can't implement `IntoIterator` for Environment if I allow a `From<T>`. It's conflicted.

Reply:
Could you elaborate on how these conflict?  I feel I'm missing something.

Reply:
here's a playground: https://play.rust-lang.org/?gist=ee3c0aa240a596f1538aee9abb31a892&version=stable

Reply:
Interesting.

- `Environment` implements `IntoIterator<(OsString, OsString)>`
- `From<IntoIterator<EnvironmentItem>>` with `EnvironmentItem` being implemented on `(OsString, OsString)`, effectively making it `From<IntoIterator<(OsString, OsString)>>`
- This somehow gets turned into implementing `From<Environment>` on `Environment`which conflicts with an impl in core.

Thats a bit round about.

Reply:
Oh, right, I got mixed up on the use case.

The main reason this was implemented was for assert_cli's `pub fn with_env<E: Into<Environment>>(mut self, env: E) -> Self`
https://github.com/killercup/assert_cli/blob/master/src/assert.rs#L164

To accomplish the same goal, without the special case `From`s this added, I wonder if we could switch `with_env` to use `FromIterator`.

Reply:
Yes but it make the API more ugly, no?

```rust
with_env(&[("foo", "bar")])

becomes

with_env(&[("foo", "bar")][..])
with_env(&[("foo", "bar")].iter())
```

right?

Reply:
Brainstorming different ways of handling this:
- Don't implement `From` and require an explicit `.iter()` in assert_cli
  - Not as nice for `assert_cli`'s API
- Don't implement `IntoIterator` and use some kind of custom way of doing the same thing
  - Less interop
- Make `Environment` a builder and require an extra "build" step to create a struct that actually implements `IntoIterator`
  - built version offers little value
- Make a non-fluent (ie no `Environment`) `Variable` struct that implements `From`.  `with_env` creates this and then uses it and `self` to create the current `ChainedVariable`.
- Make a custom `IntoEnvironment` trait and implement that instead of `From`.

The last two are the most appealing to me.  Personally, I prefer the `IntoEnvironment` approach.  It is a bit more straightforward